database/database_test.go

```diff
@@ -177,6 +178,41 @@ func TestStateForDomain(t *testing.T) {
 	}
 }
 
+func TestStatesForDomains(t *testing.T) {
+	resetDB()
+
+	if err := testDB.PutState(DomainState{Name: "a.test", Status: StatusPreloaded}); err != nil {
+		t.Errorf("cannot put state %s for test StatatesForDomains", err)
```

Comment:
nit: This and the following should instead be t.Fatalf since this is a failure in the test setup rather than a failed expectation (i.e. you are not really meaning to test the PutState function).

Reply:
Done.